src/components/form/HdInput.vue

```diff
@@ -205,12 +205,16 @@ export default {
     validate() {
       if (this.required && this.isEmpty) {
         this.showError(this.t.FORM.VALIDATION.REQUIRED);
-      } else if (this.currentType === 'email' && !validateEmail(this.value)) {
-        this.showError(this.t.FORM.VALIDATION.INVALID_EMAIL);
-      } else if (this.currentType === 'date' && !validateDate(this.value)) {
-        this.showError(this.t.FORM.VALIDATION.INVALID_DATE);
-      } else if (this.customRules.length && !this.isEmpty) {
-        this.validateCustomRules();
+      } else if (!this.isEmpty) {
+        if (this.currentType === 'email' && !validateEmail(this.value)) {
+          this.showError(this.t.FORM.VALIDATION.INVALID_EMAIL);
+        } else if (this.currentType === 'date' && !validateDate(this.value)) {
+          this.showError(this.t.FORM.VALIDATION.INVALID_DATE);
+        } else if (this.customRules.length) {
+          this.validateCustomRules();
+        } else {
+          this.hideError();
+        }
```

Comment:
What about putting some returns and remove all those else's?

Reply:
https://github.com/homeday-de/homeday-blocks/pull/452/files#diff-2c5cff5f7c1fb0874035f6dc78e56bb4R222

Reply:
How about an array of objects mapping field fieldType, validationFunction, errorMessage? Like:
```
 [
{
  fieldType: 'email',
  validationFunction: validateEmail,
 errorMessage: this.t.FORM.VALIDATION.INVALID_EMAIL
}
]
```

This way we can map every validation we have in a single file (you can export it from another file) and as it grows, there is no need to add if/else. You can do a .find(), call the validation if one validation fails.
Only one showError()/hideError() would be made.

Reply:
You can even previously .filter() the field to get the validations matching the type of the field you are currently working with.
Or you can change the structure to be like this, and access the fieldType by key:
{
  'email': [
    {
       fieldType: 'email',
       validationFunction: validateEmail,
      errorMessage: this.t.FORM.VALIDATION.INVALID_EMAIL
    }
  ],
  'text': [...]
}

Reply:
@leandroinacio @drapisarda you can check my last commit, I've tried to remove some `else` clauses.

The idea from the validator map is nice but I don't see us updating it that much to justify the effort of implementing it at the moment. I can see it inside a project with more complex business rules or move it here if we "extend" this component with our "base" business rules.
If you want we can bring this topic for a further discussion about how we deal with validations and approach it differently.
Feel free to bring to team sync if you think it makes sense.

Reply:
I will add it to the team sync if you don't mind, cause I will need to implement async validations for sliders, so I will be touching this code eventually.
Code looks good to me 👍

Reply:
Ohhhh, nice one! Please, bring it to the table as it might bite you later on!